src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -164,8 +164,16 @@ public void actionPerformed(ActionEvent e) {
             UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
             null, null, "");
         if (s != null && s.length() > 0) {
-          clientContext.user.addUser(s);
-          UserPanel.this.getAllUsers(listModel);
+          //if alphanumeric
+          if(s.matches("[a-zA-Z0-9]+")) {
+            clientContext.user.addUser(s);
+            UserPanel.this.getAllUsers(listModel);
+          } else {
+            JOptionPane.showMessageDialog(UserPanel.this,
+                "User not created. Alphanumeric characters only, with no spaces. Please try again.",
+                "User Not Created",
+                JOptionPane.ERROR_MESSAGE);
+          }
```

Comment:
I know it's handled in both places, however if the user is using the GUI to interact with the program, then they need to see some sort of message popup when they enter an invalid string, because the other mechanism for checking in the backend does not give this warning (since, if this program is being accessed from the terminal this popup is not relevant)